commit/tokenprice/processor.go

```diff
@@ -89,27 +91,75 @@ func (p *processor) Outcome(
 	// If set to zero, no prices will be reported (i.e keystone feeds would be active).
 	if p.offChainCfg.TokenPriceBatchWriteFrequency.Duration() == 0 {
 		lggr.Debugw("TokenPriceBatchWriteFrequency is set to zero, no prices will be reported")
-		return Outcome{}, nil
+		return newEmptyOutcome(), nil
 	}
 
+	inflightTokenPricesOutcome := newInflightPricesOutcome(
+		prevOutcome.InflightTokenPriceUpdates, prevOutcome.InflightRemainingChecks-1)
+
 	consensusObservation, err := p.getConsensusObservation(lggr, aos)
 	if err != nil {
-		return Outcome{}, fmt.Errorf("get consensus observation: %w", err)
+		return inflightTokenPricesOutcome, fmt.Errorf("get consensus observation: %w", err)
 	}
 
 	tokenPriceOutcome := p.selectTokensForUpdate(lggr, consensusObservation)
-	lggr.Infow(
-		"outcome token prices",
-		"tokenPrices", tokenPriceOutcome,
-	)
+	lggr.Infow("outcome token prices", "tokenPrices", tokenPriceOutcome)
 
 	if len(tokenPriceOutcome) == 0 {
 		lggr.Debugw("No token prices to report")
-		return Outcome{}, nil
+		return inflightTokenPricesOutcome, nil
+	}
+
+	// Check if we have inflight token price updates.
+	if prevOutcome.HasInflightTokenPriceUpdates() {
+		return p.computeInflightPricesOutcome(lggr, consensusObservation, prevOutcome), nil
+	}
```

Comment:
Most of the logic here is identical to chainfee except that we have this check _after_ the tokens selected for update stuff is done. Maybe we should move this to prior to `selectTokensForUpdate` to be consistent? Or is there some other reason this is different than chainfee?